shim/opentelemetry-opencensus-shim/src/opentelemetry/shim/opencensus/_patch.py

```diff
@@ -0,0 +1,55 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Optional
+
+from opencensus.trace.tracer import Tracer
+from opencensus.trace.tracers.noop_tracer import NoopTracer
+
+from opentelemetry import trace
+from opentelemetry.shim.opencensus._shim_tracer import ShimTracer
+from opentelemetry.shim.opencensus.version import __version__
+
+_logger = logging.getLogger(__name__)
+
+
+def install_shim(
+    tracer_provider: Optional[trace.TracerProvider] = None,
+) -> None:
+    if tracer_provider is None:
+        tracer_provider = trace.get_tracer_provider()
+    otel_tracer = tracer_provider.get_tracer(
+        "opentelemetry-opencensus-shim", __version__
+    )
+    shim_tracer = ShimTracer(NoopTracer(), otel_tracer=otel_tracer)
+
+    def fget_tracer(self) -> ShimTracer:
+        return shim_tracer
```

Comment:
We are returning a `ShimTracer` whether the user is using a `NoOpTracer` or `ContextTracer` right? I'm assuming this means that `ShimTracer` needs to replicate the behavior for both cases?

Reply:
> We are returning a ShimTracer whether the user is using a NoOpTracer or ContextTracer right?

Yes, afaict OpenCensus chooses one of these implementations depending on the sampling decision https://github.com/census-instrumentation/opencensus-python/blob/fd064f438c5e490d25b004ee2545be55d2e28679/opencensus/trace/tracer.py#L80-L85

> I'm assuming this means that ShimTracer needs to replicate the behavior for both cases?

What do you mean by behavior here?